### DISCUSSION TITLE
Example json for cubic eos custom fluid including caloric & transport data

### DISCUSSION THREAD

**Discussion by deje0002** [CoolProp/CoolProp#2364] [Q&A]
Hi all,
I would like to add custom fluids to coolprop. So far, adding a custom cubic eos via a json file in python works fine. From my (very basic) understanding I also need to add empirical caloric data (e.g. Cp0), also transport properties would be great. 

Does anybody have a link to a more complete example json file, ideally for a "classic" species like methane oder propane including caloric and transport property data?

All the best
Jens
PS: thanks to contributors for the great tool coolprop!

**Comment by ibell** [comment]
Well, the cubic EOS will not give you transport properties (but you could for instance use an entropy scaling approach to get a reasonable approximation). And I think CoolProp also does not allow for adding the ideal-gas Helmholtz part either.

But, the library that I developed in my day job allows that combination and many other things: https://github.com/usnistgov/teqp/

**Comment by deje0002** [comment]
Dear Ian,
regarding caloric data: I had the feeling that an ideal-gas Helmholtz was not used from the json-new-cubic-fluid approach, but was expecting an error on my side. Transport properties are prio II, but in the end one needs it all.
Thank you for the link to teqp - will have a look and try it!
Have a nice weekend
